src/metrics/rules/active_ruleset.go

```diff
@@ -704,98 +745,3 @@ func (a int64Asc) Less(i, j int) bool { return a[i] < a[j] }
 type matchRollupTargetOptions struct {
 	generateRollupID bool
 }
-
-type ruleMatchResults struct {
```

Comment:
The intermediate result structures and logic disappears now that merging is happening on the MatchResult.